### ISSUE TITLE
dvc exp show: outputs a binary file in macOS

### ISSUE THREAD

**Issue by tomasrene** [iterative/dvc#6968]
# Bug Report

## Issue name
dvc exp show: outputs a binary file in macOS

## Description
The output of dvc exp show says it's a binary file so the results is not human-readable.

### Reproduce

1. following the example in https://dvc.org/doc/start/experiments
2. dvc exp run
3. dvc exp show

### Expected

Viewing the table of run experiments on the console.

The results is:
"/var/folders/yp/dn9q0y817r35lf25w27qd8500000gn/T/tmptxifmh04/pydoc.out" may be a binary file.  See it anyway?

### Environment information

DVC version: 2.8.3 (brew)
---------------------------------
Platform: Python 3.9.7 on macOS-11.6-x86_64-i386-64bit
Supports:
        azure (adlfs = 2021.10.0, knack = 0.9.0, azure-identity = 1.7.0),
        gdrive (pydrive2 = 1.10.0),
        gs (gcsfs = 2021.11.0),
        webhdfs (fsspec = 2021.11.0),
        http (aiohttp = 3.8.0, aiohttp-retry = 2.4.6),
        https (aiohttp = 3.8.0, aiohttp-retry = 2.4.6),
        s3 (s3fs = 2021.11.0, boto3 = 1.17.106),
        ssh (sshfs = 2021.11.0),
        webdav (webdav4 = 0.9.3),
        webdavs (webdav4 = 0.9.3)
Cache types: reflink, hardlink, symlink
Cache directory: apfs on /dev/disk1s1s1
Caches: local
Remotes: None
Workspace directory: apfs on /dev/disk1s1s1
Repo: dvc, git

**Comment by karajan1001** [comment]
I can reproduce it. This is the result of my m1 MacBook pro.
```
dvc exp run
'data/images.tar.gz.dvc' didn't change, skipping
Stage 'extract' didn't change, skipping
Running stage 'train':
> python3 src/train.py
ERROR: failed to reproduce 'dvc.yaml': failed to run: python3 src/train.py, exited with -4
```

There are two problems here.
1. it failed.
2. it didn't give any useful information. 

When we run the underlying training program, it would show the failure reason.
```
$ python3 src/train.py
[1]    89999 illegal hardware instruction  python3 src/train.py
```

So I guess we need:
1. some update on https://github.com/iterative/example-dvc-experiments for m1 MacBook pro
2. Better error handling for errors like this.

**Comment by pmrowla** [comment]
@karajan1001 I think you may have replied to the wrong issue?

---

It looks like this was already resolved in discord, but for reference, the binary file output issue is a problem with the environment locale. `exp show` output is in UTF-8, so if your locale is set to a non-UTF-8 encoding, you will see the "may be a binary file" warning (since the UTF-8 characters aren't valid in whatever other encoding is being used)

Basically, the `LANG` and `LC_ALL` env vars need to be set to a UTF-8 locale to suppress the warning (and to be able to view the table properly in your terminal)

**Comment by karajan1001** [comment]
> @karajan1001 I think you may have replied to the wrong issue?

I misunderstood it, let me create a new ticket for my problem.

**Comment by daavoo** [comment]
> Basically, the `LANG` and `LC_ALL` env vars need to be set to a UTF-8 locale to suppress the warning (and to be able to view the table properly in your terminal)

Should we catch this beforehand with `getlocale` and raise more explicit error ?? 

Is there any downside of altering the users' locale with `setlocale` in our code?

**Comment by dberenbaum** [comment]
Hasn't been reported again, so closing as stale
